meps/mep-cross-references.md

```diff
@@ -0,0 +1,360 @@
+
+#### `text`
+
+If the `text` is not included, it will be filled in by the default of the target.
+
+- If the reference target is enumerated this will by default be "Sec 2.3"; "Fig. 1" for enumerated figures; "(5)" for equations; "Thm. 2", "Lemma 3", etc. depending on the target type. This can be customized in project configuration or frontmatter, however, the details of that are out of scope for this MEP.
+- If the reference target is not enumerated, the title will be used. This may be the section header text (including syntax/style) or a figure caption.
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`{number}` and `{name}`)
+
+- Enumeration (`{number}`)
+
+  - Any reference target that is enumerated can reference that number or string with a `{number}`; the `{number}` is only the enumerator and does not include any preceding text.
+    - For example, if referencing `#my-section` with default text of `Section 2.1.2`, you can change the reference to `See 2.1.2` with `[See {number}](#my-section)`.
+  - If a `{number}` is used and the node is not enumerated, the `{number}` will be replaced by "??" and a warning raised.
+
+- Reference text (`{name}`)
+  - Any node can include the text of a reference including any styles (e.g. Sections are the section title; Figures and Tables are the caption).
+  - If `{name}` is used and the node does not have an explicit name or title, the node reference label will be used.
+
+In both cases, the template can be escaped with a preceding backslash, that is `\{number}` or `\{name}`, and the text will not be replaced.
+
+#### `link`
+
+The links are defined by a scheme, which can be standard protocols (`http:`, `mailto:`).
+Here we propose three new schemes, `path` and `project` which is an extensibility point described by [CommonMark](https://spec.commonmark.org/0.30/#example-598).
+These schemes are used to indicate that the link should be resolved by MyST specific logic, and follows standard [URI][uri] syntax:
+
+```text
+URI = scheme ":" pathname ["?" query] "#" fragment
+```
+
+In most cases, as seen in the summary above the scheme is optional and can be inferred safely by the context.
+The exception is when _explicitly_ referring to an external MyST site, Jupyter Book or Sphinx documentation site.
+These URIs can be safely and easily parsed by any common URL parser. For example in Javascript:
+
+```js
+const url = new URL('project:target.md#my-ref');
+url.protocol; // "project:"
+url.pathname; // "target.md"
+url.hash; // "#my-ref"
+```
+
+For most internally linked references, we expect the inline syntax to be most commonly used, with the autolink and scheme to only be used when specific behavior is intended. The following links and references are supported:
```

Comment:
It is of note that one key use-case for auto-links, that has been requested a number of times, is that they integrate better with non-myst renderers, i.e. here `[](#target)` will show (nothing) [](#target), but  `<project:#target>` will show  <project:#target>

Reply:
I was going to suggest also having something like `[{default}](#link)`, but I guess the auto-link covers it?

Reply:
I have removed all but the end of that line:

> The following links and references are supported:

I think auto link does cover the case you are referring to.

Reply:
Marking this thread as resolved, I think that the changes fixed this up?